server/webapp/WEB-INF/rails.new/webpack/models/roles/roles.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Stream               = require('mithril/stream');
+const s                    = require('string-plus');
+const Mixins               = require('models/mixins/model_mixins');
+const PluginConfigurations = require('models/shared/plugin_configurations');
+const Routes               = require('gen/js-routes');
+const Validatable          = require('models/mixins/validatable_mixin');
+const CrudMixins           = require('models/mixins/crud_mixins');
+const _                    = require('lodash');
+
+const Roles = function (data) {
+  Mixins.HasMany.call(this, {
+    factory:    Roles.Role.create,
+    as:         'Role',
+    collection: data,
+    uniqueOn:   'name'
+  });
+  Mixins.HasUUID.call(this);
+};
+
+Roles.API_VERSION = 'v1';
+
+CrudMixins.Index({
+  type:     Roles,
+  indexUrl: Routes.apiv1AdminSecurityRolesPath(),
+  version:  Roles.API_VERSION,
+  dataPath: '_embedded.roles'
+});
+
+
+Roles.Role = function (type, data) {
+  const role                 = this;
+  role.constructor.modelType = 'role';
+
+  role.name   = Stream(s.defaultToIfBlank(data.name, ''));
+  role.type   = Stream(type);
+  role.parent = Mixins.GetterSetter();
+  role.etag   = Mixins.GetterSetter();
+  role.errors = ErrorsFromJSON(data);
+
+  role.isPluginRole = function () {
+    return role.type() === 'plugin';
+  };
+
+  role.changeRoleType = function (newType) {
```

Comment:
I see this method used from the view, wondering if role should expose something like this or if this can be handled in the view.